content/api/commands/session.md

```diff
@@ -0,0 +1,273 @@
+  - returns `false`: Cypress will execute the `setupFn` to generate a new session to use.
+- `exclude` (`object`) - contains properties `cookies` and `localStorage` for specifying items that you don't want Cypress to preserve.
+
+See [SessionOptions]()
+
+### Yields
+
+`session` yields session data (`SessionData`), which is an `object` with properties `cookies` and `localStorage`. See [SessionData](#sessiondata).
+
+## Examples
+
+### Define a Session
+
+#### Name & Setup Function
+
+The `name` should represent the kind of session you're defining. For example, if you're setting up a session for a user with read-only permissions, then you might choose the name "read-only".
+The contents of `setupFn` should be the steps to generate the session you want to preserve for later use.
+
+```js
+cy.session('read-only', () => { signIn('Adam', 'Pa$$w0Rd') })
+
+function signIn(username, password) {
+  cy.intercept('POST', '/auth').as('signIn')
+  cy.visit('/sign-in')
+  cy.get('.username').type(username)
+  cy.get('.password').type(password)
+  cy.contains('.button', 'Sign In').click()
+  cy.wait('@signIn')
+})
+```
+
+#### SessionOptions
+
+The third and last argument you can optionally pass to `session` is `SessionOptions`.
+
+### `validate` (`Function`)
+
+```js
+cy.session(
+  'read-only',
+  () => {
+    signIn('Adam', 'Pa$$w0Rd')
+  },
+  {
+    validate: () => {
+      return cy.request('GET', '/profile').should((xhr) => {
+        expect(xhr.status).to.equal(200)
+      })
+    },
+  }
+)
+```
+
+Data can become invalidated over time. For example, a user may only be authenticated for a finite duration after which the user must authenticate again. This expiration or invalidation can result in flaky tests. To remedy this, Cypress invokes `validate` on the session before using it. If the session is valid (`validate` returns `true`), then it will be used as is. Otherwise, if the session is invalid (`validate` returns `false`), then Cypress will execute the setup function first.
+
+### `exclude` (`object`)
+
+By default, Cypress preserves cookies and browser storage _in their entirety_ when the setup function executes. If data needs to be excluded, you can specify it in the `cookies` or `localStorage` property accordingly.
+
+```js
+cy.session(
+  'read-only',
+  () => {
+    signIn('Adam', 'Pa$$w0Rd')
+  },
+  {
+    exclude: {
+      cookies: [],
+      localStorage: [],
+    },
+  }
+)
+```
+
+An exclusion can be expressed as either an `object`, `string` or `RegExp`. You can specify an array of exclusions or just one.
+
+```js
+{
+  cookies: {
+    domain: 'yahoo.com'
+  }
+}
+```
+
+```js
+{
+  cookies: [
+    {
+      domain: 'yahoo.com',
+    },
+    {
+      domain: 'google.com',
+    },
+  ]
+}
+```
+
+#### `cookies` (`sessionExcludeFilterCookies`)
+
+```ts
+type sessionExcludeFilterCookies =
+  | {
+      domain?: string | RegExp
+      name?: string | RegExp
+    }
+  | string
+  | RegExp
```

Comment:
Would be nice to show examples of string and RegExp exclusions

Reply:
Yep, I'm working on that.